src/nemos/glm.py

```diff
@@ -886,9 +865,20 @@ def initialize_solver(
         # validate input
         self._validate(X, y, init_params)
 
-        self._solver_init_state, self._solver_update, self._solver_run = (
-            self.regularizer.instantiate_solver(self._predict_and_compute_loss)
-        )
+        # if using ProximalGradient, pass prox op
+        if self.solver_name == "ProximalGradient":
+            self._solver_init_state, self._solver_update, self._solver_run = (
+                super().instantiate_solver(self._predict_and_compute_loss,
+                                           self.regularizer.regularizer_strength,
+                                           prox=self.regularizer.get_proximal_operator(),
+                                           )
+            )
+
+        else:
+            self._solver_init_state, self._solver_update, self._solver_run = (
+                super().instantiate_solver(self._predict_and_compute_loss)
+            )
+
```

Comment:
I think this can be made more generic. 

Any model will have an objective function, that we call "_predict_and_compute_loss", any model will pass a loss function to a solver, so this operation is not GLM specific. It may be worth making "_predict_and_compute_loss" an abstract method of base_regressor, and we can move thus part of the code away. We can have a dedicated function, potentially in a new private module that grabs solver and the appropriate arguments (prox etc.). This function will be called directlty from base_regressor. 

In the future, this module can be used to instantiate more complex solvers (like ProjectedGradient, for example, that requires other arguments). 

Does this make sense? what do you think @clewis7 

PS I am not sure that I like the name "predict_and_compute_loss" fir it however, we will always have a predict method because we are following sklearn conventions, so that's maybe fine

Reply:
I think making `_predict_and_compute_loss()` abstract would be good because and model will have it, maybe we change the name to `objective_function()`? @BalzaniEdoardo thoughts?

Reply:
For now, I abstracted away the `instantiate_solver()` code back to the `BaseRegressor` but I think because not every solver needs the same thing it would be helpful to have a separate module for handling getting the arguments, etc. like you mentioned above (would be cleaner anyways)